### DISCUSSION TITLE
recommended place to change input depend on another input value

### DISCUSSION THREAD

**Discussion by ws-rush** [react-hook-form/react-hook-form#11118] [General]
In my code I use  onChange to chnage another value input depend on my input like following

```js
<Controller name='name' render={(field) => {
  <input type="text" onChange={({ value }) => {
    field.onChange(value)
    setValue('fullname', value)
  }} />
}} /> 
```

another way is 

```js
useEffect(() => {
  watch((data, { name, type}) => {
    if (name === 'name' && type === 'change')
      setValue(('fullname', data[value])
  })
}, [watch])

<Controller name='name' render={(field) => {
  <input type="text" onChange={field.onChange}} />
}} /> 
```

I prefer the scond method coz it seprate change controller value from change dependent values, but it need more code, which prefered method in your community

**Comment by Moshyfawn** [comment]
As with everything in software, "it depends". There's a fantastic breakdown of each use case and approach on the React docs ["You Might Not Need an Effect"](https://react.dev/learn/you-might-not-need-an-effect#chains-of-computations) page that goes into detail about when to update your state in an even handler vs. `useEffect`.
